core/src/main/java/org/apache/iceberg/deletes/DeleteGranularity.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.deletes;
+
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * An enum that represents the granularity of deletes.
+ *
+ * <p>Under partition granularity, delete writers are allowed to group deletes for different data
+ * files into one delete file. This strategy tends to reduce the total number of delete files in the
+ * table. However, it may lead to the assignment of irrelevant deletes to some data files during the
+ * job planning. All irrelevant deletes are filtered by readers but add extra overhead, which can be
+ * mitigated via caching.
+ *
+ * <p>Under file granularity, delete writers always organize deletes by their target data file,
+ * creating separate delete files for each referenced data file. This strategy ensures the job
+ * planning does not assign irrelevant deletes to data files. However, it also increases the total
+ * number of delete files in the table and may require a more aggressive approach for delete file
+ * compaction.
+ *
+ * <p>Currently, this configuration is only applicable to position deletes.
+ *
+ * <p>Each granularity has its own benefits and drawbacks and should be picked based on a use case.
+ * Despite the chosen granularity, regular delete compaction remains necessary. It is also possible
+ * to use one granularity for ingestion and another one for table maintenance.
+ */
+public enum DeleteGranularity {
```

Comment:
what is the granularity currently ? *file* ?  what is the impact to flink writers ?

Reply:
The current behavior is partition granularity. The new default will match the existing behavior.

There is no immediate impact on Flink writes. Equality deletes can only be written with partition granularity at the moment. That said, we should make Flink write position deletes with file granularity no matter what to solve the concurrent data compaction issue.

Reply:
Flink uses the old writer API right now. We will follow up to change that.